spipe/main.c

```diff
@@ -212,6 +214,16 @@ main(int argc, char * argv[])
 		exit(1);
 	}
 
+	/* Wait for threads to finish. */
+	if ((rc = pthread_join(threads[0], NULL)) != 0) {
+		warn0("pthread_join: %s", strerror(rc));
+		goto err2;
+	}
+	if ((rc = pthread_join(threads[1], NULL)) != 0) {
+		warn0("pthread_join: %s", strerror(rc));
+		goto err2;
+	}
+
```

Comment:
These will end up calling `sock_addr_freelist(sas_t);` if they fail.  Is that correct?

Reply:
fixed in #224.

Reply:
Or rather, fix *proposed* in #224.

Reply:
Or rather, fix *proposed* in #224.